File: puppeteer/pup.js
const puppeteer = require('puppeteer');
const fs = require('fs');

function delay(time) {
   return new Promise(function(resolve) { 
       setTimeout(resolve, time)
   });
}

(async () => {
  const browser = await puppeteer.launch( { defaultViewport: { width: 1200, height: 600 }, headless: false } );
  const page = await browser.newPage();

  await page.tracing.start({ screenshots: true, path: 'trace.json' /*, categories: [ "disabled-by-default-devtools.screenshot" ]*/ });
  // await page.goto('file:///C:/Users/7UR7L3/Documents/MEGAsync/dev/projects/reddit2video/Reddit2Video/cssAnim.html', { timeout: 60000 });
  await page.goto('file:///C:/Users/Jacob/Desktop/Reddit2Video/cssAnim.html', { timeout: 60000 });
  await delay( 20000 );
  await page.tracing.stop();

  // --- extracting data from trace.json ---
  const tracing = JSON.parse(fs.readFileSync('./trace.json', 'utf8'));
  const traceScreenshots = tracing.traceEvents.filter(x => (
      x.cat === 'disabled-by-default-devtools.screenshot' &&
      x.name === 'Screenshot' &&
      typeof x.args !== 'undefined' &&
      typeof x.args.snapshot !== 'undefined'
  ));
  console.log( traceScreenshots.length )
  traceScreenshots.forEach(function(snap, index) {
    fs.writeFile('frames/trace-screenshot-'+("00000"+index).substr(-5)+'.png', snap.args.snapshot, 'base64', function(err) {
      if (err) {
        console.log('writeFile error', err);
      }
    });
  });
  // --- end extracting data from trace.json ---

  await page.close();
})();

File: pup.py
import asyncio
from pyppeteer import launch
import time
import imageio
import os

async def main():
	browser = await launch(headless=False)
	page = await browser.newPage()
	await page.setViewport({'width': 1920, 'height': 1080 })
	await page.goto('file:///C:/Users/Jacob/Desktop/Reddit2Video/cssAnim.html')
	audio_len = 1 # 10 sec
	time = 0
	frame = 0
	while time < audio_len:
		frame_num = str(frame).zfill(4)
		frame += 1
		time += (1/60)
		print(time)
		try:
			await page.screenshot({'path': f'frames/example_{frame_num}.png'})
		except:
			break
	await browser.close()


asyncio.get_event_loop().run_until_complete(main())

filenames = [ "frames/" + img for img in os.listdir( "frames" ) if img.endswith( ".png" ) ]
with imageio.get_writer('movie.mp4', mode='I', fps=60) as writer:
    for filename in filenames:
        image = imageio.imread(filename)
        writer.append_data(image)